pageserver/src/tenant/timeline/walreceiver/connection_manager.rs

```diff
@@ -601,6 +604,20 @@ impl WalreceiverState {
                                     },
                                 });
                             }
+                            // If we have a candidate with the same commit_lsn as the current one, which is in the same AZ as pageserver,
+                            // and the current one is not, switch to the new one.
+                            if self.availability_zone.is_some()
+                                && existing_wal_connection.availability_zone
+                                    != self.availability_zone
+                                && Some(&new_safekeeper_broker_data.availability_zone)
```

Comment:
According to docs, seems like default protobuf 3 value for missing string is empty string, not None? I wonder why this compiles then... seems like prost considers string to be non scalar type.

Reply:
You mean default value in protobuf (wire) semantics or in Rust client library? Anyway, I tried to refresh my protobuf knowledge and found this: https://protobuf.dev/programming-guides/field_presence/

It seems that they added `optional` to `proto3` not so long ago (v3.15), and explicit presence (clear separation of empty string and None) will work only if:
- you use `optional` keyword
- `protoc` is modern at both sides
- code generation library (prost in our case) supports it

Otherwise empty string and missing value should be treated as the same.

Reply:
Ah, I misread the code, so `new_safekeeper_broker_data.availability_zone` is not optional, which conforms to the docs.